class/PostTypes/CptBase.php

```diff
@@ -103,10 +114,29 @@ public static function rewrite_post_permalink($permalink, $post)
 
   public static function rewrite_permalink($resource, $post)
   {
+    $template_variables = [
+      "{{PassleShortcode}}" => "example",
+    ];
+
     $post_shortcode = get_post_meta($post->ID, $resource->get_meta_shortcode_name(), true);
     $post_slug = get_post_meta($post->ID, $resource->get_meta_slug_name(), true);
 
-    $post_permalink_prefix = static::get_permalink_prefix();
-    return home_url($post_permalink_prefix . "/$post_shortcode/$post_slug");
+    switch ($resource->name_singular) {
+      case "post":
+        $template_variables["{{PostShortcode}}"] = $post_shortcode;
+        $template_variable["{{PostSlug}}"] = $post_slug;
+        break;
+      case "person":
+        $template_variables["{{PersonShortcode}}"] = $post_shortcode;
+        $template_variable["{{PersonSlug}}"] = $post_slug;
```

Comment:
Is this right? It seems to be equaling PersonShortcode and PersonSlug to post shortcode and post slug variables

Reply:
`CptBase` is a generic class that deals with both posts and people. 'Post' here is is referring to a native WordPress post, as both posts and people are stored as WordPress posts under separate custom post types.